packages/core/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -277,6 +279,21 @@ protected void getSentryAndroidOptions(
       options.getReplayController().setBreadcrumbConverter(new RNSentryReplayBreadcrumbConverter());
     }
 
+    // Exclude Dev Server and Sentry Dsn request from Breadcrumbs
+    String dsn = getURLFromDSN(rnOptions.getString("dsn"));
+    String devServerUrl = rnOptions.getString("devServerUrl");
+    options.setBeforeBreadcrumb(
+        (breadcrumb, hint) -> {
+          Object urlObject = breadcrumb.getData("url");
+          String url = urlObject instanceof String ? (String) urlObject : "";
+          if ("http".equals(breadcrumb.getType())
+              && ((dsn != null && url.contains(dsn))
+                  || (devServerUrl != null && url.contains(devServerUrl)))) {
```

Comment:
i: We could also only check the start of the string here too on the native side.
```suggestion
              && ((dsn != null && url.startsWith(dsn))
                  || (devServerUrl != null && url.startsWith(devServerUrl)))) {
```